content/docs/source-git/work-with-source-git/_index.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "Working with source-git"
+date: 2020-06-24T23:59:59+01:00
+draft: false
+weight: 3
+bookCollapseSection: true
+---
+
+# Working with source-git
+
+This document is focused on how packit treats a source-git repo.
+
+If you are looking for the design concept of source-git, please head on here:
+[docs/source-git-design]({{< ref "/docs/source-git/design.md" >}})
+
+The source-git concept is fairly simple - take an upstream repo and
+shove downstream content on top of it. Easy right?
+
+Packit is then able to work with such a repo and:
+
+* Create SRPMs
+
+* Sync the state into dist-git
+
+* And you are able to utilize all features of packit-service: RPM builds of PRs,
+  tests on those builds, and getting feedback in pull (merge) requests.
+
+Let's describe all of these in detail.
+
+
+## Upstream git history or just an unpacked tarball?
+
+Your source-git repo can contain upstream git history if you want. Or it
+doesn't. The alternative is to unpack a tarball corresponding to an upstream
+release. This is really up to you and packit is able to work with both.
+
+The key ingredient here is the config option `upstream_ref`. It tells packit
+the git ref which is supposed to match the archive specified in the spec file.
+All commits on top of this ref are treated either as:
+
+* downstream patches
+* downstream packaging
+
+For downstream patches, packit creates a patch file for each commit on top of
+`upstream_ref`. Downstream packaging changes are ignored in patch files since
+they can be directly committed to dist-git. You can control the ignore
+mechanism with config option `patch_generation_ignore_paths`.
+
+
+### Example
+
+Let's have a look at source-git repo for pacemaker package:
+
+    $ git log --online
+    bd12722 (HEAD -> c8s-test, origin/c8s-test, master) downstream packaging
+    7201e28 Refactor: controller: remove unused argument
+    91c557c Refactor: controller: convert active_op_t booleans to bitmask
+    2e90063 Refactor: controller: rename struct recurring_op_s to active_op_t
+    3ee90ce (tag: source-git-start) pacemaker-2.0.3 base
+
+We have 5 commits:
+* bd12722 - top commit, contains a spec file so packit won't create a patch file out of this one
+* 7201e28 - the 3rd downstream patch
+* 91c557c - the 2nd downstream patch
+* 2e90063 - the 1st downstream patch
+* 3ee90ce - the bottom commit, equals to an unpacked upstream tarball
+
+packit.yaml has only two lines:
+
+    $ cat .packit.yaml
+    specfile_path: SPECS/pacemaker.spec
+    upstream_ref: source-git-start
+
+As you can see, we are telling packit, that commit tagged as `source-git-start`
+indicates the barrier between upstream and downstream, in our case, it's just a
```

Comment:
Not sure this belongs here, but couldn't this start be indicated in some other way? Cannot this be the name of the upstream branch, for example? Where I would like to get with this: could we remove the requirement of creating a tag to mark the border between upstream and downstream and use, for example, only the name of the upstream branch?

Reply:
This is an old example which I did for Stream 8 and just moved it here, I'm gonna redo this section using a real upstream tag, that's a good point.